pilot/pkg/xds/sds.go

```diff
@@ -51,8 +52,14 @@ type SecretResource struct {
 
 var _ model.XdsCacheEntry = SecretResource{}
 
-func (sr SecretResource) Key() string {
-	return sr.SecretResource.Key() + "/" + sr.pkpConfHash
+func (sr SecretResource) Type() string {
+	return model.SDSType
+}
+
+func (sr SecretResource) Key() any {
+	h := hash.New()
+	h.Write([]byte(sr.SecretResource.Key() + "/" + sr.pkpConfHash))
```

Comment:
I thought the whole point of this change was to NOT hash the SDS type? We cannot safely hash because a collision will be a CVE

Reply:
Ok, i can revert the original impl